### ISSUE TITLE
[Auto] FileNotFoundError on Windows

### ISSUE THREAD

**Issue by soutrik7771** [Nixtla/neuralforecast#526]
# Cuurently I am showcasing the error with the example fgiven in the doc, facing the same issue with my office data which is in exact same format as expected by the module

Y_df = pd.read_parquet('https://datasets-nixtla.s3.amazonaws.com/m4-hourly.parquet')
# subsizing for quick experiment
max_len = 10 * 24
Y_df = Y_df.groupby("unique_id").tail(
    10 * 24
)  # Select last 10 days of data to make example faster
# ensure ds ends at the same point, avoid errors in cv
Y_df["ds"] = Y_df.groupby("unique_id")["ds"].transform(
    lambda x: range(len(x) - max_len + 1, max_len + 1)
)

nf = NeuralForecast(
    models=[
        AutoNHITS(
        h=ft_unit,
            loss=SMAPE(),
            num_samples=11,
            cpus=int(os.cpu_count() - 1),
            verbose=True,
        ),
    ],
    freq="H",
)

nf.fit(df=Y_df)

### Getting this chain of error from every thread opened during parallel processing---

(train_tune pid=15316) Global seed set to 4
2023-04-16 06:04:30,676	ERROR trial_runner.py:1062 -- Trial train_tune_0342e_00000: Error processing event.
ray.exceptions.RayTaskError(NotImplementedError): ray::ImplicitFunc.train() (pid=15316, ip=10.231.1.6, repr=train_tune)
  File "/anaconda/envs/pnl_py37/lib/python3.9/site-packages/ray/tune/trainable/trainable.py", line 368, in train
    raise skipped from exception_cause(skipped)
  File "/anaconda/envs/pnl_py37/lib/python3.9/site-packages/ray/tune/trainable/function_trainable.py", line 337, in entrypoint
    return self._trainable_func(
  File "/anaconda/envs/pnl_py37/lib/python3.9/site-packages/ray/tune/trainable/function_trainable.py", line 654, in _trainable_func
    output = fn()
  File "/anaconda/envs/pnl_py37/lib/python3.9/site-packages/ray/tune/trainable/util.py", line 406, in _inner
    return inner(config, checkpoint_dir=None)
  File "/anaconda/envs/pnl_py37/lib/python3.9/site-packages/ray/tune/trainable/util.py", line 398, in inner
    return trainable(config, **fn_kwargs)
  File "/anaconda/envs/pnl_py37/lib/python3.9/site-packages/neuralforecast/common/_base_auto.py", line 28, in train_tune
    model.fit(dataset, val_size=val_size, test_size=test_size)
  File "/anaconda/envs/pnl_py37/lib/python3.9/site-packages/neuralforecast/common/_base_windows.py", line 569, in fit
    trainer.fit(self, datamodule=datamodule)
  File "/anaconda/envs/pnl_py37/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 520, in fit
    call._call_and_handle_interrupt(
  File "/anaconda/envs/pnl_py37/lib/python3.9/site-packages/pytorch_lightning/trainer/call.py", line 44, in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)
  File "/anaconda/envs/pnl_py37/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 559, in _fit_impl
    self._run(model, ckpt_path=ckpt_path)
  File "/anaconda/envs/pnl_py37/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 883, in _run
    _verify_loop_configurations(self)
  File "/anaconda/envs/pnl_py37/lib/python3.9/site-packages/pytorch_lightning/trainer/configuration_validator.py", line 38, in _verify_loop_configurations
    __verify_train_val_loop_configuration(trainer, model)
  File "/anaconda/envs/pnl_py37/lib/python3.9/site-packages/pytorch_lightning/trainer/configuration_validator.py", line 86, in __verify_train_val_loop_configuration
    raise NotImplementedError(
NotImplementedError: Support for `validation_epoch_end` has been removed in v2.0.0. `NHITS` implements this method. You can use the `on_validation_epoch_end` hook instead. To access outputs, save them in-memory as instance attributes. You can find migration examples in https://github.com/Lightning-AI/lightning/pull/16520.

### adding the full trace for the error
[neuradlforecst error.txt](https://github.com/Nixtla/neuralforecast/files/11241379/neuradlforecst.error.txt)

**Comment by soutrik7771** [comment]
Urgent help is required as we are planing to create an integrated forecast framework for our organization using Nixtla.
Kindly provide us the necessary help

**Comment by FedericoGarza** [comment]
hey @soutrik7771! Thanks for using neuralforecast. 

The problem arises due to the latest `torch` and `lightning` versions. We've fixed the issue, and it is available in the `main` branch of the library. You can install it using the following: 

```
pip install "git+https://github.com/nixtla/neuralforecast.git@main"
```

If you want the use the latest version available in pip, you can fix the `torch` version using `pip install neuralforecast "torch<2.0.0"`

**Comment by luchungi** [comment]
Hi @FedericoGarza, 
Thanks for working on this. I just tried creating a fresh env and installing neuralforecast from the main branch as you suggested. However, I get the same error as in #515 i.e. RuntimeError: No best trial found for the given metric: loss.
I am running the TFT notebook: https://colab.research.google.com/github/Nixtla/neuralforecast/blob/main/nbs/examples/Forecasting_TFT.ipynb

**Comment by soutrik7771** [comment]
Hi luguchi , kindly let me know once you the error resolved , I will try after  that

**Comment by soutrik7771** [comment]
hi @luchungi  please keep me posted once you get the problem solved

**Comment by DKorman** [comment]
Hi guys, tnx for the effort! For me installing from the main branch does not solve this issue. I haven't pinned any versions of torch lib though.

**Comment by luchungi** [comment]
The error seems to stem from how the logging directory is created. Turning off the capture, I get exceptions of the flavor:
FileNotFoundError: [Errno 2] No such file or directory: b'[C:\\pathto\\ray_results\\train_tune_2023-04-20_15-07-40\\train_tune_7f3f4_00000_0_check_val_every_n_epoch](file:///C://Users//luchungi//ray_results//train_tune_2023-04-20_15-07-40//train_tune_7f3f4_00000_0_check_val_every_n_epoch)=100,hidden_size=512,hist_exog_list=spx_log_ret,input_size=20,learning_rate=0.0012_2023-04-20_15-07-43\\lightning_logs\\version_0\\events.out.tfevents.1681974471.Desktop-DC.30528.0'

I am using Windows and the issue seems to be the '=' signs in the directory name. Is there a way to customise the logging dir and subdir names?

**Comment by rnjv** [comment]
> hey @soutrik7771! Thanks for using neuralforecast.
> 
> The problem arises due to the latest `torch` and `lightning` versions. We've fixed the issue, and it is available in the `main` branch of the library. You can install it using the following:
> 
> ```
> pip install "git+https://github.com/nixtla/neuralforecast.git@main"
> ```
> 
> If you want the use the latest version available in pip, you can fix the `torch` version using `pip install neuralforecast "torch<2.0.0"`

Just here to report that the main branch fails with the same validation_epoch_end error when using NHITS.

**Comment by tg2k** [comment]
I've also run into the `FileNotFoundError` on Windows, and with `pytorch-lightning` 2.0.7. I don't believe it's the `=` character as @luchungi suggested, as I was able to create a file with that character. Instead it looks like an issue of overrunning Windows file system length limits. https://github.com/ray-project/ray/issues/29586 talks about this issue in more depth. Unfortunately the environment variable workaround there did not work for me (I tried it first in Python and then by setting it directly in Windows and restarting VSCode).

Setting the `trial_dirname_creator` under Windows to something shorter does get past this particular problem. I'll create a PR for that change.
